content/articles/web-automation-with-selenium/index.md

```diff
@@ -0,0 +1,107 @@
+
+![browser open image](/engineering-education/web-automation-with-selenium/image1.png)
+
+Next, we will be inputting a search keyword into the search field in the Google website. To do that we will have to get the search field element by inspecting the page. to inspect the page, right-click on the Google website page and click on inspect. The browser will open a window like shown in the image below
+
+![inspect browser](/engineering-education/web-automation-with-selenium/image2.png)
+
+There are different locator strategies to locate elements from the inspect element window this includes;
+- find_element_by_id 
+- find_element_by_name
+- find_element_by_class name
+- find_element_by_xpath
+
+We will be using xpath to get our elements in this guide but you can choose to use any of your choices. To get the element hover on the div tags and keep opening the one that highlights the search bar inclusive until you find the one that highlights only the search field then right-click on it, then click on copy and then click on copy xpath and paste it somewhere. Next, add the below lines to your code
+
+```python
+searchField = driver.find_element_by_xpath('/html/body/div[1]/div[3]/form/div[1]/div[1]/div[1]/div/div[2]/input')
+searchField.send_keys('university')
+
+searchField.submit()
+```
+From the above code snippet, we saved the element in a variable `searchField` we copied the xpath we copied from the inspect window and pasted it in that position i.e. into the parenthesis of the find element method.
+
+send_keys() is used to input text in selenium so the code snippet `searchField.send_keys('university')` is sending the keyword university into the search box.
+The code snippet `searchField.submit()` submit the search request.
+NOTE: you can also search for the submit button if the site you are searching from has one and use the click() method on it. But the submit() method makes it easier.
+Your complete code will look like the snippet below.
+```python
+from selenium import webdriver
+
+driver = webdriver.Chrome()
+
+driver.get('https://www.google.com/')
+
+searchField = driver.find_element_by_xpath('/html/body/div[1]/div[3]/form/div[1]/div[1]/div[1]/div/div[2]/input')
+searchField.send_keys('university')
+
+searchField.submit()
+
+
+```
+If you run your code it will open your browser, request for the Google webpage, input the university keyword in the search box and click to search.
+
+### Conclusion
+
+In conclusion, we were able to write a python script that can perform a Google search. The few lines of code from the code snippet above can be used to do the following;
+- go to any url
+- find any Html element
+- fill any form 
+- submit any form
+Link to github repo [here](https://github.com/wobin1/google-search-automation).
+
+### Further reading
+- [Selenium with python](https://selenium-python.readthedocs.io/)
+- [modern web automation with python](https://realpython.com/modern-web-automation-with-python-and-selenium/)
+
+
```

Comment:
Add these articles too
- https://www.section.io/engineering-education/getting-started-with-selenium-automation-testing/
- https://www.section.io/engineering-education/automation-testing-in-selenium/